### ISSUE TITLE
Incompatible pip's dependency

### ISSUE THREAD

**Issue by kivinblue1** [DrewThomasson/VoxNovel#34]
System: Windows 11
OS: WSL Ubuntu
Errors while installing Ubuntu-install.sh:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tts 0.21.3 requires gruut[de,es,fr]==2.2.3, but you have gruut 2.4.0 which is incompatible.
tts 0.21.3 requires numpy==1.22.0; python_version <= "3.10", but you have numpy 1.26.4 which is incompatible.
booknlp 1.0.7.1 requires transformers<=4.30.0,>=4.11.3, but you have transformers 4.40.2 which is incompatible.
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
styletts2 0.1.6 requires gruut<3.0.0,>=2.3.4, but you have gruut 2.2.3 which is incompatible.
styletts2 0.1.6 requires librosa<0.11.0,>=0.10.1, but you have librosa 0.10.0 which is incompatible.
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tts 0.21.3 requires transformers>=4.33.0, but you have transformers 4.30.0 which is incompatible.
styletts2 0.1.6 requires gruut<3.0.0,>=2.3.4, but you have gruut 2.2.3 which is incompatible.
styletts2 0.1.6 requires librosa<0.11.0,>=0.10.1, but you have librosa 0.10.0 which is incompatible.
styletts2 0.1.6 requires transformers<5.0.0,>=4.36.0, but you have transformers 4.30.0 which is incompatible.
```

**Comment by DrewThomasson** [comment]
Ignore it,

It'll probs still work

**Comment by DrewThomasson** [comment]
I was lazy with dependencies so they all stack weirdly but still work lol, 

Might fix in the future,

but considering it hasn't caused any issues so far probs not lol

**Comment by kivinblue1** [comment]
> Ignore it,
> 
> It'll probs still work

Not working in my case rn. Can you send your ubuntu/conda pip list?

**Comment by DrewThomasson** [comment]
Sure once I get to a windows computer, 

So what's the error you get when you try to run it?

**Comment by kivinblue1** [comment]
> Sure once I get to a windows computer,
> 
> So what's the error you get when you try to run it?

when starting the main .py file, a window for selecting the book file appears, after that nothing happens

**Comment by DrewThomasson** [comment]
So you give it the ebook file, and nothing happens? 

It just closes out of it? No errors or anything in the WSL terminal shell?? 🤔


You're running the gui_run.py file and NOT the 1CPU_Book_processing.py file right?

**Comment by kivinblue1** [comment]
> Итак, вы передаете ему файл электронной книги, и ничего не происходит?
> 
> Он просто закрывается и останавливает скрипт?? Никаких ошибок или чего-либо еще в терминальной оболочке WSL?? 🤔
> 
> Вы запускаете файл gui_run.py, а НЕ файл 1CPU_Book_processing.py, верно?

I'm tried run gui_run.py & 1CPU_Book_processing.py and in any case programm creating file of my book "Jacques Derrida LES EDITIONS DE MINUIT De la grammatologie.fb2" in .txt format "Jacques Derrida LES EDITIONS DE MINUIT De la grammatologie.txt". And then nothing happens:
`All required booknlp files are present. No action needed.
Low Vram mode turned on :  cpu
using device cpu
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /home/mails/nltk_data...
[nltk_data]   Package averaged_perceptron_tagger is already up-to-
[nltk_data]       date!
1% Converting input to HTML...
InputFormatPlugin: FB2 Input running
on /home/mails/Jacques Derrida LES EDITIONS DE MINUIT De la grammatologie.fb2
Parsing all content...
Forcing index.xhtml into XHTML namespace
Generating default TOC from spine...
34% Running transforms on e-book...
Merging user specified metadata...
Detecting structure...
Auto generated TOC with 0 entries.
Flattening CSS and remapping font sizes...
Source base font size is 12.00000pt
Removing fake margins...
Cleaning up manifest...
Trimming unused files from manifest...
Creating TXT Output...
67% Running TXT Output plugin
Converting XHTML to TXT...
TXT output written to /home/mails/Jacques Derrida LES EDITIONS DE MINUIT De la grammatologie.txt
Output saved to   /home/mails/Jacques Derrida LES EDITIONS DE MINUIT De la grammatologie.txt
{'pipeline': 'entity,quote,supersense,event,coref', 'model': 'big'}
/home/mails/miniconda/envs/VoxNovel/lib/python3.10/site-packages/transformers/modeling_utils.py:463: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  return torch.load(checkpoint_file, map_location="cpu")
/home/mails/miniconda/envs/VoxNovel/lib/python3.10/site-packages/booknlp/english/entity_tagger.py:22: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  self.model.load_state_dict(torch.load(model_file, map_location=device))
/home/mails/miniconda/envs/VoxNovel/lib/python3.10/site-packages/booknlp/english/bert_qa.py:20: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  self.model.load_state_dict(torch.load(modelFile, map_location=device))
/home/mails/miniconda/envs/VoxNovel/lib/python3.10/site-packages/booknlp/english/litbank_coref.py:19: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  self.model.load_state_dict(torch.load(modelFile, map_location=device))
--- startup: 5.595 seconds ---
File Working_files/temp.epub has been removed.
1% Converting input to HTML...
InputFormatPlugin: TXT Input running
on /home/mails/Jacques Derrida LES EDITIONS DE MINUIT De la grammatologie.txt
Language not specified
Creator not specified
Building file list...
Normalizing filename cases
Rewriting HTML links
Forcing index.html into XHTML namespace
34% Running transforms on e-book...
Merging user specified metadata...
Detecting structure...
Auto generated TOC with 0 entries.
Flattening CSS and remapping font sizes...
Source base font size is 12.00000pt
Removing fake margins...
Cleaning up manifest...
Trimming unused files from manifest...
Creating EPUB Output...
67% Running EPUB Output plugin
Splitting markup on page breaks and flow limits, if any...
        Looking for large trees in index.html...
        Found large tree #0
        Split into 16 parts
Generating default cover
QStandardPaths: wrong permissions on runtime directory /run/user/1000/, 0755 instead of 0700
This EPUB file has no Table of Contents. Creating a default TOC
EPUB output written to /home/mails/VoxNovel/Working_files/temp.epub
Output saved to   /home/mails/VoxNovel/Working_files/temp.epub
Created directory: Working_files/temp_ebook
/home/mails/miniconda/envs/VoxNovel/lib/python3.10/site-packages/ebooklib/epub.py:1410: FutureWarning: This search incorrectly ignores the root element, and will be fixed in a future version.  If you rely on the current behaviour, change it to './/xmlns:rootfile[@media-type]'
  for root_file in tree.findall('//xmlns:rootfile[@media-type]', namespaces={'xmlns': NAMESPACES['CONTAINERNS']}):
Saved chapter: Working_files/temp_ebook/chapter_0.txt
Saved chapter: Working_files/temp_ebook/chapter_1.txt
Saved chapter: Working_files/temp_ebook/chapter_2.txt
Saved chapter: Working_files/temp_ebook/chapter_3.txt
Saved chapter: Working_files/temp_ebook/chapter_4.txt
Saved chapter: Working_files/temp_ebook/chapter_5.txt
Saved chapter: Working_files/temp_ebook/chapter_6.txt
Saved chapter: Working_files/temp_ebook/chapter_7.txt
Saved chapter: Working_files/temp_ebook/chapter_8.txt
Saved chapter: Working_files/temp_ebook/chapter_9.txt
Saved chapter: Working_files/temp_ebook/chapter_10.txt
Saved chapter: Working_files/temp_ebook/chapter_11.txt
Saved chapter: Working_files/temp_ebook/chapter_12.txt
Saved chapter: Working_files/temp_ebook/chapter_13.txt
Saved chapter: Working_files/temp_ebook/chapter_14.txt
Saved chapter: Working_files/temp_ebook/chapter_15.txt
[nltk_data] Downloading package punkt to /home/mails/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
--- spacy: 16.710 seconds ---`

**Comment by DrewThomasson** [comment]
# That's weird, also send me your ebook file for me to test on my end when I get to a windows computer.

The .txt and .fb2 one.

Don't worry I'm still working on getting you the conda pip freeze thing,

- There is also a zip folder named, `Example_Working_files.zip` in the VoxNovel git repo , This contains very small ebook files to test on VoxNovel, This should allow you to see if the issue is just the ebook file your giving it.
- 
``` bash
Example_Working_files.zip
```




------------------------------------------------------------------------------------------------------------




# Also I noticed your book is not in English?

- That could potentially be the issue, but your book is French so that uses the same character system so probably not.🤷

- Booknlp (The thing that's used to figure out which characters said what lines, Is only guaranteed to work on English at this moment,  ) :(
- IT MIGHT work tho??? It kind of works in Spanish I have no idea how, I'm not sure about French but it's worth a shot to try I guess.
- But The Default (text to speech) being (XTTS): is multilingual though.


#### I do have other audiobook generator projects that **are multilingual** though:
- Only difference is they all just make a normal audiobook with one voice
- 
- They also all do have docker implementations of them with a web gui 👀

This one sounds the best and uses XTTS, has voice cloning ,and is multilingual.
https://github.com/DrewThomasson/ebook2audiobookXTTS

- This one is super fast, is multilingual, But does NOT have voice cloning. (it uses Piper's)
- Also the voices are like Siri level quality, so doesn't sound as good
https://github.com/DrewThomasson/ebook2audiobookpiper-tts

------------------------------------------------------------------------------------------------------------


# Potential things you could try while we wait on me getting to a windows computer: -->

# 1. Also I have a headless(terminal only) implementation of VoxNovel running in a free google colab as well for you to test if you want to try that.
- https://colab.research.google.com/drive/15pp2hFBo2fD3legDQfWY5DMt-aI-HKKF?usp=sharing




------------------------------------------------------------------------------------------------------------

# 2. ALSO I do have a backup .tar image of a working Ubuntu wsl env with VoxNovel already pre-installed. (You don't have to do this, I'm giving you this because you seem like your a wsl power user )

- I planned on using it for the windows install but, ran into issues with getting the sound driver working when importing it.

- But it should work 

- You don't have to do this just also putting this out there cause you seem like you know what your doing in WSL.

https://huggingface.co/drewThomasson/VoxNovel_WSL_ENV/blob/main/Windows_WSL_VoxNovel.tar

#### If you end up trying this here are some scripts I made that should have the import command in them.

- I don't remember if they work or not, but chatgpt should be Able to help you with this if you end up doing this.

- This one was used for automatically importing the wsl .tar image. 
https://github.com/DrewThomasson/VoxNovel/blob/main/shell_install_scripts/Windows-install-scripts/wsl_import_run.sh

- This one was used for launching the imported WSL env image. 
- https://github.com/DrewThomasson/VoxNovel/blob/main/shell_install_scripts/Windows-install-scripts/import_wsl_env.ps1
#### The important commands in that launch wsl file are these two at the top of it.
``` bash
# Switch to the appropriate user
su - drew
export DISPLAY=:0
```

**Comment by DrewThomasson** [comment]
# Update:

## Just tried the normal installation on a windows computer I have. 

### Seems to work fine for me.


It might be your ebook or how much ram you have idk.

Could be that your ebook isn't English idk

**Comment by kivinblue1** [comment]
> # Update:
> ## Just tried the normal installation on a windows computer I have.
> ### Seems to work fine for me.
> 
> It might be your ebook or how much ram you have idk.
> 
> Could be that your ebook isn't English idk
> 
> Hope that helps at all

Yes, the book contains Cyrillic

**Comment by DrewThomasson** [comment]
> > # Update:
> > ## Just tried the normal installation on a windows computer I have.
> > ### Seems to work fine for me.
> > It might be your ebook or how much ram you have idk.
> > Could be that your ebook isn't English idk
> > Hope that helps at all
> 
> Yes, the book contains Cyrillic

That might be whats breaking it, unrecognized characters being thrown into booknlp. :/
